webdriver/tests/bidi/browsing_context/locate_nodes/locator.py

```diff
@@ -261,7 +261,89 @@ async def test_locate_by_context(bidi_session, inline, top_context, domain):
             "type": "node",
             "sharedId": any_string,
             "value": {
-                "attributes": {"src": any_string},
+                "attributes": {"src": any_string, "id": "target"},
+                "childNodeCount": 0,
+                "localName": "iframe",
+                "namespaceURI": "http://www.w3.org/1999/xhtml",
+                "nodeType": 1,
+            }
+        }
+    ]
+
+    recursive_compare(expected, result["nodes"])
+
+
+@pytest.mark.parametrize("domain", ["", "alt"], ids=["same_origin", "cross_origin"])
+@pytest.mark.asyncio
+async def test_locate_by_context_in_iframe(bidi_session, inline, top_context, domain):
+    iframe_url_2 = inline("<div>foo</div>", domain=domain)
+    iframe_url_1 = inline(f"<div><iframe id='target' src='{iframe_url_2}'></iframe></div>")
+    page_url = inline(f"<iframe src='{iframe_url_1}'></iframe>")
+
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=page_url, wait="complete"
+    )
+
+    contexts = await bidi_session.browsing_context.get_tree(root=top_context["context"])
+    iframe1_context = contexts[0]["children"][0]
+    iframe2_context = contexts[0]["children"][0]["children"][0]
+
+    result = await bidi_session.browsing_context.locate_nodes(
+        context=iframe1_context["context"],
+        locator={"type": "context", "value": { "context": iframe2_context["context"] }}
+    )
+
+    expected = [
+        {
+            "type": "node",
+            "sharedId": any_string,
+            "value": {
+                "attributes": {"src": any_string, "id": "target"},
+                "childNodeCount": 0,
+                "localName": "iframe",
+                "namespaceURI": "http://www.w3.org/1999/xhtml",
+                "nodeType": 1,
+            }
+        }
+    ]
+
+    recursive_compare(expected, result["nodes"])
+
+
+@pytest.mark.parametrize("domain", ["", "alt"], ids=["same_origin", "cross_origin"])
+@pytest.mark.parametrize("mode", ["open", "closed"])
+@pytest.mark.asyncio
+async def test_locate_by_context_in_shadow_dom(bidi_session, inline, top_context, domain, mode):
+    iframe_url_1 = inline(f"<div>foo</div>", domain=domain)
```

Comment:
You could use the `get_test_page` fixture to all the setup basically in one line.